extras/ssd1306/ssd1306.h

```diff
@@ -0,0 +1,93 @@
+/* vim: set expandtab ts=4 sw=4 : */
+#ifndef _SSD1306__H_
+#define _SSD1306__H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <errno.h>
+
+// shifted
+#define SSD1306_I2C_ADDR	(0x3C)
+
+/* SSD1306 commands */
+#define SSD1306_SET_MEM_ADDR_MODE   (0x20)
+#define SSD1306_ADDR_MODE_HORIZ     (0x0)
+#define SSD1306_ADDR_MODE_VERT      (0x1)
+#define SSD1306_ADDR_MODE_PAGE      (0x2)
+
+#define SSD1306_SET_COL_ADDR        (0x21)
+#define SSD1306_SET_PAGE_ADDR       (0x22)
+#define SSD1306_SET_DISP_START_LINE (0x40)
+#define SSD1306_SET_CONTRAST        (0x81)
+#define SSD1306_SET_SEGMENT_REMAP0  (0xA0)
+#define SSD1306_SET_SEGMENT_REMAP1  (0xA1)
+#define SSD1306_SET_ENTIRE_DISP_ON  (0xA5)
+#define SSD1306_SET_ENTIRE_DISP_OFF (0xA4)
+#define SSD1306_SET_INVERSION_OFF   (0xA6)
+#define SSD1306_SET_INVERSION_ON    (0xA7)
+
+#define SSD1306_SET_MUX_RATIO       (0xA8)
+#define SSD1306_MUX_RATIO_MASK      (0x3F)
+#define SSD1306_SET_DISPLAY_OFF     (0xAE)
+#define SSD1306_SET_DISPLAY_ON      (0xAF)
+#define SSD1306_SET_SCAN_DIR_FWD    (0xC0)
+#define SSD1306_SET_SCAN_DIR_BWD    (0xC8)
+#define SSD1306_SET_DISPLAY_OFFSET  (0xD3)
+#define SSD1306_SET_OSC_FREQ        (0xD5)
+#define SSD1306_SET_PRE_CHRG_PER    (0xD9)
+
+#define SSD1306_SET_COM_PINS_HW_CFG (0xDA)
+#define SSD1306_COM_PINS_HW_CFG_MASK    (0x32)
+#define SSD1306_SEQ_COM_PINS_CFG    (0x02)
+#define SSD1306_ALT_COM_PINS_CFG    (0x12)
+#define SSD1306_COM_LR_REMAP_OFF    (0x02)
+#define SSD1306_COM_LR_REMAP_ON     (0x22)
+
+#define SSD1306_SET_DESEL_LVL       (0xDB)
+#define SSD1306_SET_NOP             (0xE3)
+
+#define SSD1306_SET_CHARGE_PUMP     (0x8D)
+#define SSD1306_CHARGE_PUMP_EN      (0x14)
+#define SSD1306_CHARGE_PUMP_DIS     (0x10)
```

Comment:
Does user need those definition to use the library? Can they be moved into .c file to make header shorter? So the header  would contain only library interface information.